### DISCUSSION TITLE
Sub Graph Using Java

### DISCUSSION THREAD

**Discussion by amit-kumaryadav** [JanusGraph/janusgraph#3147] [Q&A]
Hi I'm using the janusgraph v0.6.2 with java application. When I'm trying the subgraph query I'm getting **Error during serialization**. 

**Java code: -** 
--      Object graph =   g.V().hasLabel("Template").repeat(__.outE("hasDevice")
                .subgraph("sg").inV()).cap("sg").next();

**Error:-**

org.apache.tinkerpop.gremlin.driver.exception.ResponseException: Error during serialization: java.lang.IllegalStateException

Using below remote-objects.yaml file configuration. 

hosts: [localhost]
port: 8182
serializer: {
  className: org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1,
  config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] },
}

**Comment by vtslab** [comment]
Serialization happens on Gremlin Server, so Gremlin Server needs to be configured to use the TinkerIoRegistryV3d0 in order to be able to serialize the TinkerGraph instance returned by subgraph(). For deserialization to succeed, your remote-object.yaml also needs the additional 
TinkerIoRegistryV3d0.
More details can be found in an older discussion:
[https://lists.lfaidata.foundation/g/janusgraph-users/topic/79936118#2044 ](https://lists.lfaidata.foundation/g/janusgraph-users/topic/79936118#2044  )
Best wishes,   Marc

**Reply by vtslab** [reply]
See also: [https://github.com/JanusGraph/janusgraph/issues/2843](https://github.com/JanusGraph/janusgraph/issues/2843)
